### DISCUSSION TITLE
发币的这三种能力BurnCapability，FreezeCapability，MintCapability 结构体都是空，为啥还可以当做这3个操作的参数

### DISCUSSION THREAD

**Discussion by hi-chenerge** [NonceGeek/Move-Camp#263] [Q&A] [answered]
initialize 调用的 initialize_internal 方法最后返回这3个能力
`(BurnCapability<CoinType>{ }, FreezeCapability<CoinType>{ }, MintCapability<CoinType>{ })`
这里面啥属性也没有，然后 mint 的时候居然还要传入这个，不理解这是啥意义?
`public fun mint<CoinType>( amount: u64, _cap: &MintCapability<CoinType>)`

我看源码`_cap`这个参数压根就没用，那为啥要传呢？

**Comment by hotboice** [comment, marked as answer]
fun mint设计的入参需要传入mint_cap。由于struct只能在定义它的module中创建，init后mint_cap等这三种能力保存在account下，也就代表只有拥有mint_cap的account，才可以调用mint。可以简单理解为一个权限标记

**Comment by qiwihui** [comment]
也可以参考 discussion 的这个问题： https://github.com/NonceGeek/Web3-dApp-Camp/discussions/250

**Comment by hi-chenerge** [comment]
由于struct只能在定义它的module中创建， 这句话是最核心的，确实在我的module 中不能创建 MintCapability
